text/0000-trits-t5b1/0000-trits-t5b1.md

```diff
@@ -0,0 +1,119 @@
++ Feature name: `trits-t5b1`
++ Start date: 2019-09-25
++ RFC PR: [iotaledger/bee-rfcs#13](https://github.com/iotaledger/bee-rfcs/pull/13)
++ Bee issue: [iotaledger/bee#61](https://github.com/iotaledger/bee/issues/61)
+
+# Summary
+
+This RFC proposes an encoding that packs `5` `balanced trits` to a single `signed byte`. It may be referred to as `5t/1b`, `5t1b`, `t5/b1`, `t5b1`, `5/1`, `10/2`, ... To comply with most languages identifier rules, we'll prefer using `t5b1`.
+
+<!-- TODO Update with link to RFC#10 -->
+This encoding implements the interface presented by RFC#10.
+
+# Motivation
+
+A `byte` is composed of `8` `bits` that can hold `2^8 = 256` different values. On the other hand, `6` `trits` can hold `3^6 = 729` values while `5` `trits` can hold `3^5 = 243` values. The maximum number of `trits` a `byte` can hold is then `5`.
+
+As this is the most memory-efficient way to encode `trits` in `bytes`, `t5b1` is often used for network communications to increase throughput and for storage layers to reduce memory consumption.
+
+# Detailed design
+
+For efficient conversions, we can make use of a LookUp Table (`LUT`).
+
+We give the `t5b1` `LUT` as a reference for implementation, it contains the `243` possible values ordered like this: `0`, `1`, ... `120`, `121`, `-121`, `-120`, ... `-2`, `-1`.
```

Comment:
The `LUT` should cover all `256` possible cases for values of a byte. To the byte values in range `243..255` we can assign some default mapping to 5 trits, eg. 5 zero trits (ie. 5 zero trits can be encoded in several ways). This way we can avoid runtime checks (branching on each iteration of a loop can degrade performance in case of false branch prediction). Anyway, all data checks must be cryptographic, ie. using hash, MAC or sig. Additionally, we can have two variants of "decode from bytes" function: 1. checked & slow (checks for correct values for each byte and returns false in case of incorrect value) and, 2. fast & unchecked.

Reply:
Good point!